src/aosm/azext_aosm/cli_handlers/onboarding_core_vnf_handler.py

```diff
@@ -0,0 +1,211 @@
+from azext_aosm.configuration_models.onboarding_vnf_input_config import (
+    OnboardingCoreVNFInputConfig,
+)
+from azext_aosm.configuration_models.common_parameters_config import (
+    CoreVNFCommonParametersConfig,
+)
+from azext_aosm.definition_folder.builder.bicep_builder import (
+    BicepDefinitionElementBuilder,
+)
+from azext_aosm.definition_folder.builder.json_builder import (
+    JSONDefinitionElementBuilder,
+)
+from azext_aosm.inputs.arm_template_input import ArmTemplateInput
+from azext_aosm.inputs.vhd_file_input import VHDFileInput
+from azext_aosm.common.utils import render_bicep_contents_from_j2, get_template_path
+
+from .onboarding_vnf_handler import OnboardingVNFCLIHandler
+logger = get_logger(__name__)
+
+
+class OnboardingCoreVNFCLIHandler(OnboardingVNFCLIHandler):
+    """CLI handler for publishing NFDs."""
+
+    def _get_input_config(
+        self, input_config: Dict[str, Any] = None
+    ) -> OnboardingCoreVNFInputConfig:
+        """Get the configuration for the command."""
+        if input_config is None:
+            input_config = {}
+        return OnboardingCoreVNFInputConfig(**input_config)
+
+    def _get_params_config(
+        self, config_file: dict = None
+    ) -> CoreVNFCommonParametersConfig:
+        """Get the configuration for the command."""
+        with open(config_file, "r", encoding="utf-8") as _file:
+            params_dict = json.load(_file)
+        if params_dict is None:
+            params_dict = {}
+        return CoreVNFCommonParametersConfig(**params_dict)
+
+    def _get_processor_list(self) -> [BaseInputProcessor]:
+        """Get the list of processors."""
+        processor_list = []
+        # for each arm template, instantiate arm processor
+        for arm_template in self.config.arm_templates:
+            arm_input = ArmTemplateInput(
+                artifact_name=arm_template.artifact_name,
+                artifact_version=arm_template.version,
+                default_config=None,
+                template_path=Path(arm_template.file_path).absolute(),
+            )
+            processor_list.append(
+                AzureCoreArmBuildProcessor(arm_input.artifact_name, arm_input)
+            )
+
+        # Instantiate vhd processor
+        if not self.config.vhd.artifact_name:
+            self.config.vhd.artifact_name = self.config.nf_name + "-vhd"
+        vhd_processor = VHDProcessor(
+            name=self.config.vhd.artifact_name,
+            input_artifact=VHDFileInput(
+                artifact_name=self.config.vhd.artifact_name,
+                artifact_version=self.config.vhd.version,
+                default_config=self._get_default_config(self.config.vhd),
+                file_path=Path(self.config.vhd.file_path).absolute(),
+                blob_sas_uri=self.config.vhd.blob_sas_url,
+            ),
+        )
+        processor_list.append(vhd_processor)
+        return processor_list
+
+    def build_base_bicep(self) -> BicepDefinitionElementBuilder:
```

Comment:
I think this method might actually be identical to Nexus handler. Worth deduplicating?